go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -53,7 +53,7 @@ def dependency_from_details(details)
         Dependency.new(
           name: details["Path"],
           version: version,
-          requirements: details["Indirect"] || dependency_is_replaced(details) ? [] : reqs,
+          requirements: dependency_is_replaced(details) ? [] : reqs,
```

Comment:
This doesn't 100% fit in with Dependabot's model. The details for a direct or indirect dependency are the same in the `go.mod` so we do have "requirements" for it. However, Dependabot treats a Dependency with requirements as "top_level?" which usually means a direct dependency.

Not supplying requirements for indirect dependencies will require this PR gets more complicated as there are several places that will bail out in the update process if requirements aren't present or changed.

Reply:
Just to add, this would also cause transititive dependencies to be labeled as direct in the commit metadata. That's used by some users via https://github.com/dependabot/fetch-metadata in determining further actions for a Dependabot PR.

So I think we should fix this correctly and maintain the direct/indirect distinction if possible.